File: src/lessons/welcome/bat/bool2/AnswerCellEntity.java
package lessons.welcome.bat.bool2;

import static plm.core.ValueSerializer.deserialize;
import static plm.core.ValueSerializer.serialize;

import plm.universe.bat.BatEntity;

public class AnswerCellEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(answerCell((Boolean)param[0], (Boolean)param[1], (Boolean)param[2])));
    }
  }

  /* BEGIN TEMPLATE */
  boolean answerCell(boolean isMorning, boolean isMom, boolean isAsleep)
  {
    /* BEGIN SOLUTION */
    return (!isAsleep) && !(isMorning && !isMom);
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/welcome/bat/bool2/AnswerCellEntity.scala
package lessons.welcome.bat.bool2

import plm.universe.bat.BatEntity
import plm.universe.bat.BatTest

class AnswerCellEntity extends BatEntity {

    override def run() {
      import plm.core.ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
      	setTestResult(i, serialize(answerCell(param(0).asInstanceOf[Boolean], param(1).asInstanceOf[Boolean], param(2).asInstanceOf[Boolean]) ))
	  }
	}

	/* BEGIN TEMPLATE */
	def answerCell(isMorning:Boolean, isMom:Boolean, isAsleep:Boolean):Boolean = {
	/* BEGIN SOLUTION */
   return (! isAsleep) && !(isMorning && !isMom)
	/* END SOLUTION */
	}
	/* END TEMPLATE */
}
